ietf-layer0-types.yang

```diff
@@ -892,6 +892,230 @@ module ietf-layer0-types {
+          type l0-types:dwdm-n;
+          description
+            "The central frequency of DWDM.";
+          reference
+            "RFC 6205: Generalized Labels for Lambda-Switch-Capable
+             (LSC) Label Switching Routers";
+        }
+      }
+      case cwdm {
+        leaf cwdm-n {
+          when "derived-from-or-self(../../../grid-type,
+                \"wson-grid-cwdm\")" {
+            description
+              "Valid only when grid type is CWDM.";
+          }
+          type l0-types:cwdm-n;
+          description
+            "Channel wavelength computing input.";
+          reference
+            "RFC 6205: Generalized Labels for Lambda-Switch-Capable
+             (LSC) Label Switching Routers";
+        }
+      }
+      case flexi-grid {
+        uses l0-types:flexi-grid-label-start-end;
+      }
+    }
+    reference
+      "RFC 6205: Generalized Labels for Lambda-Switch-Capable (LSC)
+       Label Switching Routers";
+  }
+
+  grouping wdm-label-step {
+    description
+      "Label step information for fixed & flexi-DWDM or CWDM grid";
+    choice l0-grid-type {
+      description
+        "Grid type: DWDM, CWDM, etc.";
+      case fixed-dwdm {
+        leaf wson-dwdm-channel-spacing {
+          when "derived-from-or-self(../../grid-type,
+                \"wson-grid-dwdm\")" {
+            description
+              "Valid only when grid type is DWDM.";
+          }
+          type identityref {
+            base dwdm-ch-spc-type;
+          }
+          description
+            "Label-step is the channel spacing (GHz), e.g., 100.000,
+             50.000, 25.000, or 12.500 GHz for DWDM.";
+          reference
+            "RFC 6205: Generalized Labels for Lambda-Switch-Capable
+             (LSC) Label Switching Routers";
+        }
+      }
+      case cwdm {
+        leaf wson-cwdm-channel-spacing {
+          when "derived-from-or-self(../../grid-type,
+                \"wson-grid-cwdm\")" {
+            description
+              "Valid only when grid type is CWDM.";
+          }
+          type identityref {
+            base cwdm-ch-spc-type;
+          }
+          description
+            "Label-step is the channel spacing (nm), i.e., 20 nm
+             for CWDM, which is the only value defined for CWDM.";
+          reference
+            "RFC 6205: Generalized Labels for Lambda-Switch-Capable
+             (LSC) Label Switching Routers";
+        }
+      }
+      case flexi-grid {
+        uses flexi-grid-label-step;
+      }
+    }
+    reference
+      "RFC 6205: Generalized Labels for Lambda-Switch-Capable (LSC)
+       Label Switching Routers,
+       ITU-T G.694.2 (12/2003): Spectral grids for WDM applications:
+       CWDM wavelength grid,
+       RFC 8363: GMPLS OSPF-TE Extensions in Support of Flexi-Grid
+       Dense Wavelength Division Multiplexing (DWDM) Networks";
+  }
+
+  grouping wdm-label-hop {
+    description
+      "Generic label-hop information for fixed & flexi-DWDM or
```

Comment:
what is the generic label-hop information? the quantum of spectrum? Apparently yes , since the flexible-grid case report the frequency slot information right ?

Reply:
Yes and this description is consistent with the other grouping, wson-label-hop and flexi-grid-label-hop.

Reply:
since it is new we can also think to add just few words to specify that hop in this context is related to amount of spectrum. 
Just a suggestion ...non a mandate comment

Reply:
ok